.github/workflows/update-specs-and-client-libraries.yaml

```diff
@@ -0,0 +1,136 @@
+name: update-specs-and-client-libraries
+run-name: Update specs and client libraries
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - 'generated/artifacts/**'
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+    inputs:
+      type-of-change:
+        description: 'Type of change?'
+        required: true
+        default: Patch
+        type: choice
+        options:
+          - Major
+          - Minor
+          - Patch
+      update-onfido-node:
+        description: 'Update onfido-node?'
+        required: true
+        default: false
+        type: boolean
+env:
+  GITHUB_SHA: env.GITHUB_SHA
+jobs:
+  generate_specs_and_libraries:
+    name: Build single-file OpenAPI specifications and client libraries
+    runs-on: ubuntu-latest
+    outputs:
+      typescript_axios_version: ${{ steps.generator.outputs.typescript_axios_version }}
+    container:
+      image: openapitools/openapi-generator-cli:v7.3.0
+      env:
+        OPENAPI_GENERATOR_COMMAND: docker-entrypoint.sh
+        BUMP_CLIENT_LIBRARY_VERSION: ${{ inputs.type-of-change || 'Minor' }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install pre-requisites
+        run: |
+          apt-get update
+          apt-get install -yqq \
+            gettext-base \
+            git \
+            jq
+      - name: Validate multi-file OpenAPI specification
+        run: |
+          $OPENAPI_GENERATOR_COMMAND validate -i openapi.yaml
+      - name: Generate specifications and client libraries
+        id: generator
+        run: |
+          ./shell/generate.sh
+      - name: Store generated artifacts and finalization scripts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-${{ github.workflow }}-${{ github.run_id }}-${{ github.run_number }}
+          path: |
+            generated/artifacts
+            generators/*/exclusions.txt
+  update_specs:
+    name: Update and commit single-file OpenAPI specifications
+    runs-on: ubuntu-latest
+    needs: generate_specs_and_libraries
+    if: github.event_name == 'push'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GH_ACTION_ACCESS_TOKEN }}
+          ref: ${{ github.ref_name }}
+      - uses: actions/download-artifact@v4
+        with:
+          name: artifacts-${{ github.workflow }}-${{ github.run_id }}-${{ github.run_number }}
+      - name: Commit and push single-file OpenAPI specifications
+        run: |
+          if [ -z "$(git status --porcelain=v1 generated/artifacts/openapi-*)" ];
+          then
+            echo "no change detected"
+          else
+            echo "changes detected"
+            git config user.name "GitHub Actions Bot"
+            git config user.email "<>"
+            git status generated/artifacts/openapi-*
+            git commit -m "Single file specifications refresh" generated/artifacts/openapi-*
+            git diff HEAD^ HEAD
+            # git push
```

Comment:
Just showing the diff without pushing to be sure about what's going to be pushed after merging the two Github Actions.